jobbergate-api/jobbergate_api/file_validation.py

```diff
@@ -0,0 +1,265 @@
+from jinja2 import Template, TemplateSyntaxError
+from loguru import logger
+from yaml import YAMLError, safe_load
+
+from jobbergate_api.apps.applications.schemas import ApplicationConfig
+
+
+class UploadedFilesValidationError(Buzz):
+    """Raise exception when facing any validation error on the uploaded files."""
+
+
+def log_error_and_raise_http_error(params: DoExceptParams):
+    """
+    Work with ``UploadedFilesValidationError.handle_errors`` when something goes wrong.
+
+    In case any error occurs in the context manager, the error is logged and
+    a HTTPException is raised.
+
+    :param DoExceptParams params: Params from handle_errors.
+    :raises HTTPException: Code 422, unprocessable entity.
+    """
+    logger.error(params.final_message)
+    raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail=params.final_message)
+
+
+def get_suffix(file: UploadFile) -> str:
+    """
+    Get the suffix (file extension) of an UploadFile.
+
+    :param UploadFile file: Target file.
+    :return str: File extension.
+    """
+    return PurePath(file.filename).suffix
+
+
+CheckUploadedFilesEquation = Callable[[List[UploadFile]], None]
+"""
+Type alias describing the function signature used to check the uploaded files.
+
+The functions are expected to raise UploadedFilesValidationError if any error is detected.
+"""
+
+check_uploaded_files_dispatch: List[CheckUploadedFilesEquation] = []
+"""List the function used to check the uploaded files."""
+
+
+def perform_all_checks_on_uploaded_files(file_list: List[UploadFile]) -> None:
+    """
+    Loop though the check-list for uploaded files specifications.
+
+    Individual checks will raise UploadedFilesValidationError if the business
+    roles are not met.
+
+    :param List[UploadFile] file_list: Upload file list.
+    """
+    logger.debug("Preparing to perform all checks on the uploaded files")
+    with UploadedFilesValidationError.handle_errors(
+        "The uploaded files are invalid",
+        do_except=log_error_and_raise_http_error,
+        re_raise=False,
+    ):
+        for check in check_uploaded_files_dispatch:
+            check(file_list)
+
+
+def register_check_uploaded_files(checker: CheckUploadedFilesEquation) -> CheckUploadedFilesEquation:
+    """
+    Use this decorator to register functions to check the uploaded files.
+
+    It creates a new entry on ``check_uploaded_files_dispatch``.
+
+    :return CheckUploadedFilesEquation: The decorated function.
+    """
+    check_uploaded_files_dispatch.append(checker)
+
+    @wraps(checker)
+    def decorator(file_list: List[UploadFile]) -> None:
+        return checker(file_list)
+
+    return decorator
+
+
+@register_check_uploaded_files
+def check_uploaded_files_extensions(file_list: List[UploadFile]) -> None:
+    """
+    Check the list of uploaded files to count the file extensions.
+
+    Raise UploadedFilesValidationError if the business roles are not met.
```

Comment:
```suggestion
    Raise UploadedFilesValidationError if the business rules are not met.
```

Reply:
looks like there are 5 occurrences of this

Reply:
Copy-paste propagating a typo, good catch. Solved in [667186b](https://github.com/omnivector-solutions/jobbergate/pull/128/commits/667186b3538f3d1c28e2f77563d2af51f8ed8756).